caikit/runtime/service_generation/protoable.py

```diff
@@ -91,6 +107,49 @@ def handle_protoables_in_union(arg_type: Type) -> Type:
     log.debug("Skipping non-protoable argument type [%s]", arg_type)
 
 
+def _make_union_list_source_type_name(args: List) -> str:
+    """Make the name for  source class that wraps the given types"""
+    return f"UnionList{''.join([get_args(arg)[0].__name__.capitalize() for arg in args])}Source"
+
+
+def make_union_dataobjects(union_protoables: List) -> Type[DataBase]:
+    """Dynamically create a union list source message type"""
+    package = "caikit_data_model.runtime"
+    cls_name = _make_union_list_source_type_name(union_protoables)
+    if cls_name not in _UNION_LIST_SOURCE_TYPES:
+        data_object_map = {}
+        data_object_list = []
+        for arg in union_protoables:
+            arg_type = get_args(arg)[0]
+            arg_name = f"{arg_type.__name__.capitalize()}Sequence"
```

Comment:
Based on @joerunde 's comments, I will create a subsequent PR in which I stick all of these `IntSequence`, `StrSequence` etc. directly into the `caikit runtime interfaces`, and reference them when they're needed

Reply:
Also `file, filelist, and directory` from `DataStreamSource`

Reply:
Resolved `#1` in this issue, opened https://github.com/caikit/caikit/issues/300 for `#2`.